service/src/main/java/uk/gov/hmcts/reform/fpl/service/LocalAuthorityUserService.java

```diff
@@ -26,27 +27,27 @@
     private final AuthTokenGenerator authTokenGenerator;
     private final IdamClient client;
     private final Set<String> caseRoles = Set.of("[LASOLICITOR]", "[CREATOR]");
+    private final RequestData requestData;
 
     @Autowired
     public LocalAuthorityUserService(OrganisationService organisationService,
                                      AuthTokenGenerator authTokenGenerator,
                                      CaseUserApi caseUserApi,
                                      IdamClient idamClient,
-                                     SystemUpdateUserConfiguration userConfig) {
+                                     SystemUpdateUserConfiguration userConfig, RequestData requestData) {
         this.organisationService = organisationService;
         this.authTokenGenerator = authTokenGenerator;
         this.caseUserApi = caseUserApi;
         this.client = idamClient;
         this.userConfig = userConfig;
+        this.requestData = requestData;
     }
 
-    public void grantUserAccessWithCaseRole(String authorisation,
-                                            String userId,
-                                            String caseId,
+    public void grantUserAccessWithCaseRole(String caseId,
                                             String caseLocalAuthority) {
-        List<String> userIds = findUserIds(authorisation, caseLocalAuthority);
+        List<String> userIds = findUserIds(requestData.authorisation(), caseLocalAuthority);
```

Comment:
Probably requestData is not required on that level, it's only used to pass it to OrganisationService and it already contains RequestData injected.

Reply:
I removed the request.authorisation() call. But I had to retain the injection of RequestData, as grantUserAccessWithCaseRole() uses requestData.userId().